doc/read-the-docs-site/doc/marconi-as-a-library/tutorials/write-first-indexer.rst

```diff
@@ -1,6 +1,122 @@
+   :language: haskell
+   :start-after: BLOCKSTART env
+   :end-before: BLOCKEND env
+
+With that environment type, let's create the function which runs the indexer.
+
+.. literalinclude:: BasicApp.hs
+   :language: haskell
+   :start-after: BLOCKSTART runBlockInfoIndexer
+   :end-before: BLOCKEND runBlockInfoIndexer
+
+Putting the final pieces together, we get:
+
+.. literalinclude:: BasicApp.hs
+   :language: haskell
+   :start-after: BLOCKSTART runBlockInfoListIndexer
+   :end-before: BLOCKEND runBlockInfoListIndexer
+
+In order to run that you need to start a Cardano node which will produce a UNIX
+socket file. That socket file is what Marconi will use to communicate with the
+node.
+
+Instructions on how to run a Cardano node are detailed `here <https://github.com/input-output-hk/cardano-node>`_ along with the list of Cardano networks.
+In our case, let's run a Cardano node for the preprod network (a testnet)::
+
+  cardano-node run --topology config/cardano-node/preprod/topology.json \
+      --socket-path ~/cardano-node/preprod/cardano-node.socket \
+      --config config/cardano-node/preprod/config.json \
+      --database-path ~/cardano-node/preprod/db
+
+
+You can run that with::
+
+  cabal run marconi-doc -- ~/cardano-node/preprod/cardano-node.socket 1
+
+You should get an output that looks like::
+
+  [marconi-tutorial:Info:1] [2023-10-16 12:14:26.87 UTC] The starting point for the chain-sync protocol is ChainPointAtGenesis
+  [marconi-tutorial:Info:9] [2023-10-16 12:14:26.88 UTC] Starting from ChainPointAtGenesis.
+  [marconi-tutorial:Info:9] [2023-10-16 12:14:36.88 UTC] Synchronising (2.42%). Current synced point is ChainPoint(Slot 1011389, BlockHash 797ff6324ed7049e263014d0845596b1da60699c47f74b17432580a61222166c) and current node tip is ChainTip(Slot 41775270, BlockHash b77475adc22489a33a223f94cfb85255a14afd76d6ed830d68a2a791df220734, BlockNo 1491171). Processed 46258 blocks and 1 rollbacks in the last 10s (4626 blocks/s).
+  [marconi-tutorial:Info:9] [2023-10-16 12:14:46.88 UTC] Synchronising (4.70%). Current synced point is ChainPoint(Slot 1962692, BlockHash 960127359b2703f4055994f42e24872f8713b8e5ed6fbcca2f76704ccea07f7a) and current node tip is ChainTip(Slot 41775270, BlockHash b77475adc22489a33a223f94cfb85255a14afd76d6ed830d68a2a791df220734, BlockNo 1491171). Processed 47393 blocks and 0 rollbacks in the last 10s (4739 blocks/s).
+  [marconi-tutorial:Info:9] [2023-10-16 12:14:56.88 UTC] Synchronising (6.87%). Current synced point is ChainPoint(Slot 2871998, BlockHash 5accb16e4c9c4ee4f163720bed7fbc1883b4aeb5b30260cb051c280dac6a2e97) and current node tip is ChainTip(Slot 41775270, BlockHash b77475adc22489a33a223f94cfb85255a14afd76d6ed830d68a2a791df220734, BlockNo 1491171). Processed 45397 blocks and 0 rollbacks in the last 10s (4540 blocks/s).
+  ...
+
+SQLiteIndexer backend
+---------------------
+
+The previous code used the ``Core.ListIndexer`` which is very simple to use, but keeps all the events in memory.
```

Comment:
```suggestion
The previous code used the ``Core.ListIndexer` is very simple to use, but keeps all the events in memory.
```

Reply:
The suggestion is erronous isn't it?

Reply:
Yes, apparently I can't read (and I hadn't spotted the _simple_ in this sentence)